wit/core.wit

```diff
@@ -0,0 +1,89 @@
+package wasmcloud:core@0.1.0
+
+interface types {
+  type link-settings =          tuple<string, string>
+  type host-env-values =        tuple<string, string>
+  type actor-links =            list<link-defination>
+  type cluster-issuers =        list<cluster-issuer-key>
+  type cluster-issuer-key =     string 
+  type capability-contract-id = string
+  type blob =                   list<u8>
+  type trace-context =          tuple<string, string>
+
+  enum log-level {
+    trace,
+    debug,
+    info,
+    warn,
+    error,
+  }
+
+  record link-defination {
+    actor-id:     string,
+    provider-id:  string,
+    link-name:    string,
+    contract-id:  string,
+    values:       option<link-settings>,
+  }
+
+  record health-check-response {
+    healthy: bool,
+    message: string,
+  }
+
+  record health-check-request { }
+  
+  record host-data {
```

Comment:
host data is a JSON payload injected into a capability provider via JSON-over-stdio.. we may not need to define this as a wit record.. 🤔

Reply:
my thoughts were that this will allow for the provider to have a struct for mashalling... a la https://github.com/wasmCloud/provider-sdk-go/blob/14cfe1cb919cad4b92f2bb5d4c93c93c64a6da39/provider.go#L68

Reply:
Yeah it's confusing and not anywhere near an ideal situation... but `wit` doesn't define a serialization format, so when we use wit to generate structs and then use language-specific features to serialize those structs, we're basically tightly coupling the wit to a wire format. This probably won't hurt us to start with, but if we draw a hard line in the sand and say that we'll never define opaque wire payloads as structures to be serialized from wit contracts, it might save us in the long run .... or I could be worrying about nothing :)

Reply:
I mean, we could always add it back later if we need it.  Ill pull it out for now.  
That said, I would like to identify a place to make sure all language specific structures start in sync.

Reply:
Yeah I guess it's a tradeoff worth weighing. The cost of seemingly creating a tight coupling versus the benefit of having a single source of truth on data structures.